### ISSUE TITLE
Check failed: target_blobs.size() == source_layer.blobs_size() (1 vs. 2) Incompatible number of blobs for layer conv1

### ISSUE THREAD

**Issue by DeanDon** [BVLC/caffe#5318]
I use the resnet model finetune use my dataset,and get this error:
`F0222 17:03:37.133123  8684 net.cpp:777] Check failed: target_blobs.size() == source_layer.blobs_size() (1 vs. 2) Incompatible number of blobs for layer conv1`
same way in googlenet is work,but in resnet i get error.

**Comment by DeanDon** [comment]
This is data layer and conv1 layer in prototxt file:
`name: "ResNet-50"
layer {
    name: "data"
    type: "Data"
    top: "data"
    top: "label"
    include {
        phase: TRAIN
    }
    transform_param {
        mirror: true
        crop_size: 227
        mean_value: 104
        mean_value: 117
        mean_value: 123
    }
    data_param {
        source: "/home/gpudouble1/caffe/data/mydata/mytask_train_lmdb"
        batch_size: 16
        backend: LMDB
    }
}
layer {
    name: "data"
    type: "Data"
    top: "data"
    top: "label"
    include {
        phase: TEST
    }
    transform_param {
        mirror: false
        crop_size: 227
        mean_value: 104
        mean_value: 117
        mean_value: 123
    }
    data_param {
        source: "/home/gpudouble1/caffe/data/mydata/mytask_val_lmdb"
        batch_size: 2
        backend: LMDB
    }
}

layer {
    bottom: "data"
    top: "conv1"
    name: "conv1"
    type: "Convolution"
    convolution_param {
        num_output: 64
        kernel_size: 7
        pad: 3
        stride: 2
        weight_filler {
            type: "msra"
        }
        bias_term: false

    }
}`

**Comment by DeanDon** [comment]
This is data layer in prototxt file:
`name: "ResNet-50"
layer {
    name: "data"
    type: "Data"
    top: "data"
    top: "label"
    include {
        phase: TRAIN
    }
    transform_param {
        mirror: true
        crop_size: 227
        mean_value: 104
        mean_value: 117
        mean_value: 123
    }
    data_param {
        source: "/home/gpudouble1/caffe/data/mydata/mytask_train_lmdb"
        batch_size: 16
        backend: LMDB
    }
}
layer {
    name: "data"
    type: "Data"
    top: "data"
    top: "label"
    include {
        phase: TEST
    }
    transform_param {
        mirror: false
        crop_size: 227
        mean_value: 104
        mean_value: 117
        mean_value: 123
    }
    data_param {
        source: "/home/gpudouble1/caffe/data/mydata/mytask_val_lmdb"
        batch_size: 2
        backend: LMDB
    }
}

layer {
    bottom: "data"
    top: "conv1"
    name: "conv1"
    type: "Convolution"
    convolution_param {
        num_output: 64
        kernel_size: 7
        pad: 3
        stride: 2
        weight_filler {
            type: "msra"
        }
        bias_term: false

    }
}`

**Comment by DeanDon** [comment]
`name: "ResNet-50"
layer {
    name: "data"
    type: "Data"
    top: "data"
    top: "label"
    include {
        phase: TRAIN
    }
    transform_param {
        mirror: true
        crop_size: 227
        mean_value: 104
        mean_value: 117
        mean_value: 123
    }
    data_param {
        source: "/home/gpudouble1/caffe/data/mydata/mytask_train_lmdb"
        batch_size: 16
        backend: LMDB
    }
}
layer {
    name: "data"
    type: "Data"
    top: "data"
    top: "label"
    include {
        phase: TEST
    }
    transform_param {
        mirror: false
        crop_size: 227
        mean_value: 104
        mean_value: 117
        mean_value: 123
    }
    data_param {
        source: "/home/gpudouble1/caffe/data/mydata/mytask_val_lmdb"
        batch_size: 2
        backend: LMDB
    }
}

layer {
    bottom: "data"
    top: "conv1"
    name: "conv1"
    type: "Convolution"
    convolution_param {
        num_output: 64
        kernel_size: 7
        pad: 3
        stride: 2
        weight_filler {
            type: "msra"
        }
        bias_term: false

    }
}`


This is data layer and conv1 layer

**Comment by katerakelly** [comment]
Your source and target conv1 layers have different numbers of parameters. I suggest carefully comparing the prototxt files of your network and the pre-trained net you're copying params from.

**Comment by shelhamer** [comment]
The above is right—the pre-trained net most likely includes a bias term while your new arch sets `bias_term: false`. Please ask such usage questions on the mailing list.

From https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:

> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.

**Comment by asfix** [comment]
I removed all the batchnorm layers during the finetuning and it worked for me. The accuracy was more than 70%.  I am not sure that removing those layers hurt the performance. If someone clarifies me I ll be pleased.

**Comment by ujsyehao** [comment]
@asfix  Removing batchnorm layers hurt performance!

**Comment by Angzz** [comment]
I solved this problem by changing the conv1's ``bias_term`` to **True**. that is to say, the pretrained caffemodel must be in consistent with you own architecture defination.

I do not suggest remove the BN layer, because it wil harm the performance.
